snuba/datasets/session_events_raw.py

```diff
@@ -0,0 +1,87 @@
+from datetime import timedelta
+from typing import Mapping, Sequence
+
+from snuba.datasets.dataset import TimeSeriesDataset
+from snuba.clickhouse.columns import (
+    ColumnSet,
+    DateTime,
+    LowCardinality,
+    Nullable,
+    String,
+    UInt,
+    Float,
+    UUID,
+)
+from snuba.datasets.schemas.tables import MergeTreeSchema
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.query.extensions import QueryExtension
+from snuba.query.organization_extension import OrganizationExtension
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.prewhere import PrewhereProcessor
+from snuba.query.query_processor import QueryProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+
+
+class SessionEventsRawDataset(TimeSeriesDataset):
```

Comment:
Looking at the requirements, I think this should become the SessionDataset. Which means every raw is a session and we drop the idea of having a raw event table unless there is any functional reasons for having the raw events.

To achieve this we would use a Replacing Merge Tree table in clickhouse. This table engine merges periodically rows that have the same primary key by replacing those with an lower version with those with a new version. The version can be the event timestamp. 

When querying this table we would apply a uniq operator by session_id since the session_id could be temporarily duplicated waiting for a merge. In alternative clickhouse offer a FINAL keyword which would trigger the merge of the rows in the query, but that is likely to be worse for performance than the uniq

We also need to define a version column for the replacing merge. this has to be a single column, and it has to be either an integer or a datetime. Clickhouse uses it to define which is the last event for the session to use to replace the rest of the session.
We generally use the deleted column (0 not deleted, 1 deleted) so we can delete rows, but here we will also have to merge undeleted rows for the same session. 
Can we use the event timestamp or do we not trust that due to clock drift ? if we cannot trust it, can we use MaxInt64 as seq number for the deleted row ? and use the seq as the version column ?